### ISSUE TITLE
Suggestion: allowMorphTools gamerule

### ISSUE THREAD

**Issue by nivthefox** [Budschie/BudschieMorphMod#46]
I'd really love a game rule that allowed me to disable the use of tools (weapons, bows, anything that has a "use" action) while morphed, for balance reasons. I really don't mind people placing blocks, but right now being a Bee or Bat is so very strong that there's just no reason not to do it, which makes me kinda sad on my multiplayer server.

**Comment by AFinalWord** [comment]
Especially the blaze, that one is pretty OP

**Comment by AFinalWord** [comment]
Oh and also I like this idea, forgot to say that :)

**Comment by Budschie** [comment]
Yeah, I agree. The blaze morph especially is really OP, and I have yet to figure out a way to compensate for its strong abilities. Whilst I think that a gamerule that disables or enables the usage of tools with this mod would be a great idea (and I am certainly going to implement it in the future), it is a bit of a temporary solution. So, if you have any ideas on how to balance morphs (maybe through a cost system), feel free to put those into this issue. Also, as I am writing this, I think it would be great to add a multiplier on the harvest speed of different morphs (so, a blaze would mine half as fast, a bat would mine at a normal rate, etc.).
\- Budschie

**Comment by AFinalWord** [comment]
Maybe you could make some abilities of the morphs have XP costs, so for example: To shoot a fireball as a blaze, it would maybe be 1 whole XP level, or just whatever you think a good balance would be.

**Comment by AFinalWord** [comment]
Or maybe, to get a certain morph you have to have a certain XP level, that could work too!

**Comment by nivthefox** [comment]
> if you have any ideas on how to balance morphs (maybe through a cost system), feel free to put those into this issue.

So even if we had a better way to balance them, I'd still want this feature sooner than later while we figure out what that better way is.

Honestly, I think the problem is that Morphs is best when used in conjunction with another system.  For example, I'm using Morphs in conjunction with Mana and Artifice. The only way to get a Morph is through a manaweaving recipe I have added which gives you the morph via a datapack function.

Ideally, once you have the morph, it should cost you a low level of mana every tick just to stay in the morph, like Bound Sword, Bound Bow, Bound Axe, and Bound Shield do, plus an additional amount of mana (all configurable via configuration?) to use the abilities of the morph.

This is honestly what I'd like to see done, and I am highly tempted to write my own compatibility mod which brings this result about.   I am not sure Morph itself _can_ provide an effective way to balance, because it fundamentally lacks any concept of progression system.  That's not a failing in the mod, it just means that if you're looking for a balanced experience with Morph, some 3rd-party Mod has to be involved which provides the progression and power source.

**Comment by AFinalWord** [comment]
> > if you have any ideas on how to balance morphs (maybe through a cost system), feel free to put those into this issue.
> 
> So even if we had a better way to balance them, I'd still want this feature sooner than later while we figure out what that better way is.
> 
> Honestly, I think the problem is that Morphs is best when used in conjunction with another system. For example, I'm using Morphs in conjunction with Mana and Artifice. The only way to get a Morph is through a manaweaving recipe I have added which gives you the morph via a datapack function.
> 
> Ideally, once you have the morph, it should cost you a low level of mana every tick just to stay in the morph, like Bound Sword, Bound Bow, Bound Axe, and Bound Shield do, plus an additional amount of mana (all configurable via configuration?) to use the abilities of the morph.
> 
> This is honestly what I'd like to see done, and I am highly tempted to write my own compatibility mod which brings this result about. I am not sure Morph itself _can_ provide an effective way to balance, because it fundamentally lacks any concept of progression system. That's not a failing in the mod, it just means that if you're looking for a balanced experience with Morph, some 3rd-party Mod has to be involved which provides the progression and power source.
--------------------------------------------------------------------------------------------------------------------------------------------
While this is a great idea / feature, this doesn't feel very fit with this kind of mod, idk why, but for me it just doesn't fit, but that's up to @Budschie to figure out what's best I guess.

**Comment by Budschie** [comment]
I've now gone ahead and implemented this into the mod (partially): If you set this gamerule to false, you cannot use flint and steel or bows, and digging tools like pickaxes are just ignored (=> if you have an iron pick and are morphed, it is as if you were mining the block by hand). I decided to not put any damage nerf on swords or axes, as you'd be pretty defenseless without those.

This is currently just a really rudimentary implementation though, therefore I'm probably gonna change it when I have the time to do it.
